manifests/node/kubelet.pp

```diff
@@ -228,21 +260,31 @@
   Package <| title == 'containernetworking-plugins' |> -> Service['kubelet']
 
   if $manage_firewall {
-    firewalld_custom_service { 'kubelet':
-      ensure      => $ensure,
-      short       => 'kubelet',
-      description => 'Kubernetes kubelet daemon',
-      ports       => [
-        {
-          port     => '10250',
-          protocol => 'tcp',
-        },
-      ],
-    }
-    firewalld_service { 'Allow k8s kubelet access':
-      ensure  => $ensure,
-      zone    => 'public',
-      service => 'kubelet',
+    if $facts['firewalld_version'] and $firewall_type == 'firewalld' {
```

Comment:
This is a very bad UX, and will definitely cause trouble for people.
If a user were to provide
```
manage_firewall => true,
firewall_type => 'firewalld',
```
then this code could still deploy an entirely unwanted iptables config instead of what's been explicitly specified.

If you want to have `$firewall_type` be optional, then you should define it as `Optional[K8s::Firewall_type] $firewall_type = undef`. Otherwise explicitly specifying 'firewalld' would have the same meaning as not specifying a value - a.k.a. being okay with anything.

My suggestion;
```puppet
# In the parameter lists
  Optional[K8s::Firewall_type] $firewall_type = undef,
  # Or '= $k8s::firewall_type', which should also be an Optional[K8s::Firewall_type] = undef
)

# ...

# In the code;
if $facts['firewalld_version'] {
  $_firewall_type = pick($firewall_type, 'firewalld')
} else {
  $_firewall_type = pick($firewall_type, 'iptables')
}
case $_firewall_type {
  'firewalld' : {
    # firewalld_custom_service { ...
  }
  'iptables': {
    # include firewall
    # firewall { ...
  }
  default: { }
}
```